docs/design/services/s3/transfermanager/README.md

```diff
@@ -0,0 +1,66 @@
+# Project Tenets (unless you know better ones)
+
+1. Meeting customers in their problem space allows them to deliver value
+   quickly.
+2. Meeting customer expectations drives usability.
+3. Discoverability drives usage.
+
+# Introduction
+
+This project provides a much improved experience for S3 customers needing to
+easily perform uploads and downloads of objects to and from S3 by providing
+the S3 `TransferManager`, a high level library built on the S3 client.
+
+# Project Goals
+
+1. For the use cases it addresses, i.e. the transfer of objects to and from S3,
+   TransferManager is the preferred solution. It is easier and more intuitive
+   than using the S3 client. In the majority of situations, it is more
+   performant.
+1. TransferManager provides a truly asynchronous API that conforms to the norms
+   present in the rest of the SDK.
+1. TransferManager makes efficient use of system resources.
+1. TransferManager performs well in cold JVMs.
+1. TransferManager supplements rather than replaces the lower level S3 client.
+
+# Non Project Goals
+
+1. Ability to use the blocking, synchronous client.
+
+   Using a blocking client would severely impede the ability to deliver on goals
+   #2 and #3.
+
+# Customer-Requested Changes from 1.11.x
+
+* TransferManager supports progress listeners that are easier to use.
+
+  Ref: https://github.com/aws/aws-sdk-java-v2/issues/37#issuecomment-316218667
```

Comment:
More of a meta-question: do we know yet whether these will be S3-specific, or built into the core?

Reply:
I think it makes sense for them to be same thing. Or at least, TransferManager supports both the core and an "enhanced" TransferManager specific version if it makes sense to have one.